tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -768,3 +793,198 @@ def test_copy_obj_after_server_pod_restart(self):
+        self.version_etag[self.bucket_name].extend(resp[1])
+        LOGGER.info("Step 1: Uploaded object of %s size before enabling versioning.", self.f_size)
+        LOGGER.info("Step 2: Enable versioning on %s.", self.bucket_name)
+        resp = self.s3_ver.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 2: Enabled versioning on %s.", self.bucket_name)
+        LOGGER.info("Step 3: Upload same object %s after enabling versioning, list & verify "
+                    "the same.", self.object_name)
+        args = {'file_path': self.multipart_obj_path}
+        resp = self.ha_obj.parallel_put_object(event, self.s3_test_obj, self.bucket_name,
+                                               self.object_name, **args)
+        assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+        self.version_etag[self.bucket_name].extend(resp[1])
+        resp = self.ha_obj.list_verify_version(self.s3_ver, self.bucket_name, self.version_etag[
+            self.bucket_name])
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 3: Uploaded same object %s after enabling versioning, listed & verified "
+                    "the same.", self.object_name)
+        LOGGER.info("Step 4: Shutdown server pod with replica method and verify cluster & "
+                    "remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            pod_prefix=[const.SERVER_POD_NAME_PREFIX], delete_pod=[self.delete_pod],
+            num_replica=self.num_replica-1)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete server pod")
+        pod_name = list(resp[1].keys())[0]
+        if self.set_type == const.STATEFULSET:
+            self.set_name = resp[1][pod_name]['deployment_name']
+        elif self.set_type == const.REPLICASET:
+            self.deployment_name = resp[1][pod_name]['deployment_name']
+        self.restore_pod = True
+        self.restore_method = resp[1][pod_name]['method']
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 4: Successfully shutdown server pod %s. Verified cluster and services "
+                    "states are as expected & remaining pods status is online.", pod_name)
+        LOGGER.info("Step 5: Get object versions of %s with VersionID & verify etags.",
+                    self.object_name)
+        resp = self.ha_obj.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                               bkt_name=self.bucket_name, obj_name=self.object_name,
+                                               ver_etag=self.version_etag[self.bucket_name])
+        assert_utils.assert_true(resp[0], f"Get Object with versionID failed {resp[1]}")
```

Comment:
lets print response preferably in lib for debugging purpose

Reply:
There are already prints in core libs. That's why not added here.